mathesar/users/decorators.py

```diff
@@ -0,0 +1,30 @@
+import functools
+
+from django.contrib.auth import get_user_model
+from django.shortcuts import redirect
+from django.urls import reverse
+
+
+def superuser_must_not_exists(view_function=None):
```

Comment:
Wouldn't returning some error be more robust here? Something about using redirects as a security measure in this way makes my skin itchy.

Reply:
We are **returning** a `redirect` without going to the next step, so even if it does not get redirected (which shouldn't happen) the user will never be able to create a superuser. Also, we are using `redirects` as the strategy for our redirecting unauthenticated users to the login page, so I don't see much of a difference.